### ISSUE TITLE
No Output From "skaffold dev" If Pod Exits Quickly

### ISSUE THREAD

**Issue by cliffburdick** [GoogleContainerTools/skaffold#1106]
Hi, I have a kind=Job I'm launching that only runs for about 2 seconds, and prints about 100 lines. If I use skaffold dev, the pod starts up fine, and kubectl shows the pod status as "Completed". However, skaffold doesn't print anything after deploying:

Build complete in 9.757380217s
Starting deploy...
kubectl client version: 1.12
job.batch/pod-test created
Deploy complete in 557.612274ms

If instead I run kubectl logs on that same container, I see the 100 or so lines it printed before exiting. If I run a longer-running job with skaffold dev, the output shows as expected.

### Expected behavior
Output of pod shows
### Actual behavior
No output shows even after the pod has completed
### Information

- Skaffold version: 0.15.1
- Operating system: Ubuntu 18.04
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1alpha3
build:
  artifacts:
  - docker:
      buildArgs: {}
      dockerfilePath: deploy/docker/Dockerfile.test
    imageName: test/uncommitted
    workspace: .
  tagPolicy:
    envTemplate:
      template: '{{.IMAGE_NAME}}:{{.SALT}}'
deploy:
  kubectl:
    manifests:
    - test-f96317.yaml*
kind: Config

```

### Steps to reproduce the behavior

1) Create Job with short-lived container that prints some output
2) Run Job with skaffold dev and observe output

**Comment by balopat** [comment]
+1 with @coollog we did see this as well, thanks for filing the issue!

**Comment by cliffburdick** [comment]
@balopat any updates on this one? If you can point me to roughly where the code is that handles this, I might be able to take a look.

**Comment by dgageot** [comment]
Hi @cliffburdick, can you reproduce with skaffold 0.19?

**Comment by cliffburdick** [comment]
I will git it a try David. Thanks

**Comment by cliffburdick** [comment]
@dgageot yes, the problem still happens. My container prints 89 lines in about 50ms before completing, and skaffold showed no output on 0.19. kubectl logs shows all the output.

**Comment by cliffburdick** [comment]
@dgageot sure, I'll try that now and report back.

**Comment by cliffburdick** [comment]
@dgageot that worked! will this be merged into 0.20?

**Comment by dgageot** [comment]
Excellent, I'll try to make sure it goes in

**Comment by dgageot** [comment]
Fixed with #1344
